docs/walkthrough.md

```diff
@@ -0,0 +1,553 @@
+# A walkthrough
+
+This walkthrough will take you through setting up the CMS on an existing project, and it will introduce several core CMS concepts.
+If you prefer to learn from code than learn from a wall of text,
+you probably want to look at the companion [tiny CMS project repo](https://github.com/onespacemedia/tiny-cms-project).
+
+## Django settings
+
+We'll need to add a few settings that the CMS depends on.
+Don't worry too much about what these do for now; the concepts behind them will be explained in depth over the course of the walkthrough.
+
+First, tell us the name of your site. We depend on this in one of our [template functions](template-functions.md):
+
+```python
+SITE_NAME = 'a tiny project'
+```
+
+And your domain; it is used by template functions to turn relative /urls/ into http://actual.absolute/urls/:
+
+```python
+SITE_DOMAIN = 'example.com'
+```
+
+We don't want things set to be offline to appear in the front-end (our pages have on/offline controls), but we definitely want them to appear in our admin:
+
+```python
+PUBLICATION_MIDDLEWARE_EXCLUDE_URLS = (
+    '^admin/.*',
+)
+```
+
+Add our core CMS things to your INSTALLED_APPS:
+
+```python
+INSTALLED_APPS = [
+  # .....
+  'cms',
+  'cms.apps.pages',
+  'cms.apps.media',
+  # Links is optional, but it's very handy to have.
+  'cms.apps.links',
+]
+```
+
+Add our context processors to our template context processors (in `['OPTIONS']['context_processors']`):
+
+```python
+'cms.apps.pages.context_processors.pages',
+```
+
+Add the CMS middleware to your MIDDLEWARE:
+
+```python
+MIDDLEWARE = [
+  # ...
+  'cms.middleware.PublicationMiddleware',
+  'cms.apps.pages.middleware.PageMiddleware',
+]
+```python
+
+And finally, options for our HTML editor:
+
+```python
+WYSIWYG_OPTIONS = {
+    'height': 500,
+    'plugins': [
+        'advlist autolink link image lists charmap hr anchor pagebreak',
+        'wordcount visualblocks visualchars code fullscreen cmsimage hr',
+    ],
+    'toolbar1': 'code | cut copy pastetext | undo redo | bullist numlist | link unlink anchor cmsimage | blockquote',
+    'menubar': False,
+    'toolbar_items_size': 'small',
+    'block_formats': 'Paragraph=p;Header 2=h2;Header 3=h3;Header 4=h4;Header 5=h5;Header 6=h6;',
+    'convert_urls': False,
+    'paste_as_text': True,
+    'image_advtab': True,
+}
```

Comment:
Maybe add a link after this to TinyMCE docs so that users can find out what these all mean. Maybe also state that cmsimage is our plugin that allows for media app integration